lib/Doctrine/Common/Cache/Psr6/DoctrineProvider.php

```diff
@@ -66,6 +73,26 @@ protected function doFetch($id)
         return $item->isHit() ? $item->get() : false;
     }
 
+    /**
+     * @inheritDoc
+     */
+    protected function doFetchMultiple(array $keys): array
+    {
+        $items = $this->pool->getItems(array_map('rawurlencode', $keys));
+        if ($items instanceof Traversable) {
+            $items = iterator_to_array($items);
+        }
+
+        $items = array_combine($keys, $items);
```

Comment:
`$pool->getItems()` returns items keyed by the cache key. But it makes no guarantee that keys are in the same order.

Reply:
Fixed https://github.com/doctrine/cache/pull/375/commits/d11f9218cc83498e19e378119d57eb954d438974